### ISSUE TITLE
Add application_limits_exempt_when?

### ISSUE THREAD

**Issue by etiennecaldichouryatnapta** [alisaifee/flask-limiter#407]
Hi!

I have the same need as mentionned in [this issue](https://github.com/alisaifee/flask-limiter/issues/295): I want to exclude some requests from the application_limits.
But it doesn't seem like using a callable for specifying limits solves the issue?
Because this callback won't be called at each request? (it is the case for default_limits_exempt_when, so it would be awesome to have a application_limits_exempt_when) !
I'll check the code but if someone has an idea to make this work with the actual code it would be great!

Thanks a lot for your help!

**Comment by alisaifee** [comment]
@etiennecaldichouryatnapta : the callable that returns the application limits will in fact be called for each request.

Here's an example where you can see that if the header `X` is present no application limit will be enforced:

```python
from flask import Flask, request

from flask_limiter import Limiter
from flask_limiter.util import get_remote_address

app = Flask(__name__)


def app_limit() -> str | None:
    if "X" in request.headers:
        return None

    return "2/day"


lim = Limiter(get_remote_address, app=app, application_limits=[app_limit])


@app.route("/1")
def one() -> str:
    return "one"


if __name__ == "__main__":
    app.run()

```

**Comment by alisaifee** [comment]
I do agree though, having `application_limits_exempt_when` would be a better way to do this instead of working around with the callable. I'll add this for the next release.

**Comment by alisaifee** [comment]
This is now available in [3.4.0](https://github.com/alisaifee/flask-limiter/releases/tag/3.4.0)

**Comment by etiennecaldichouryatnapta** [comment]
Wow so fast thanks a lot 🙏
